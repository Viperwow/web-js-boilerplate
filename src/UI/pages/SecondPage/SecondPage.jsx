import {identity as _identity} from 'lodash';
import React from 'react';
import {Link} from 'react-router-dom';

const SecondPage = ({
  text,
  t = _identity,
}) => (
  <div className="page">
    {t(text)}
    <ul>
      <li>
        <Link to="/1" replace>
          1
        </Link>
      </li>
      <li>
        <Link to="/3" replace>
          3
        </Link>
      </li>
      <li>
        <Link to="/not-found" replace>
          4
        </Link>
      </li>
    </ul>
  </div>
);

export default SecondPage;

// This must be added to support HMR in dynamic imports
if (module.hot) {
  module.hot.accept();
}
