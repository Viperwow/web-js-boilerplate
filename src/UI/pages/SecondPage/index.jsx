import loadable from 'src/common/loadable';
import Loader from 'src/UI/components/Loader';

const DynamicSecondPage = loadable({
  importer: () => import('src/UI/pages/SecondPage/SecondPageConnector' /* webpackChunkName: "SecondPage" */), // eslint-disable-line max-len
  LoaderComponent: Loader,
});

export default DynamicSecondPage;
