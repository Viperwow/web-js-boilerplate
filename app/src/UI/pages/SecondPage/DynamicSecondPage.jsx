import React from 'react';
import dynamic from 'src/common/dynamic';

const DynamicSecondPage = () => {
  const Component = dynamic(
    import('src/UI/pages/SecondPage' /* webpackChunkName: "SecondPage" */),
  );

  return <Component />;
};

export default DynamicSecondPage;
