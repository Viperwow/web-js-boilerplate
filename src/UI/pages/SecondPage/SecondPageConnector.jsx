// Vendors
import i18n from 'i18next';
// Common
import connectQuery from 'src/common/connectQuery';
// GQL
import QUERY_LOCALE from 'src/queries/locale';
// UI
import SecondPage from './SecondPage';

const mapStateToProps = () => ({
  text: i18n.t('pages.second'),
});

export default connectQuery(QUERY_LOCALE, mapStateToProps)(SecondPage);
