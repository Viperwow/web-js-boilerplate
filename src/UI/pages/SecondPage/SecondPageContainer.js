import connectQuery from 'src/common/connectQuery';
import withOnError from 'src/common/withOnError';
import withTranslate from 'src/common/withTranslate';
import {compose} from 'src/helpers/utility';
import {GET_APP_LOCALE} from 'src/queries';
import SecondPage from './SecondPage';

const mapStateToProps = (_, {t}) => ({
  text: t('pages.second'),
});

export default compose(
  withTranslate,
  connectQuery(GET_APP_LOCALE, mapStateToProps),
  withOnError,
)(SecondPage);
