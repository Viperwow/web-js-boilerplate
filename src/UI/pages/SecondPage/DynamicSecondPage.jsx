// HoC
import loadable from 'src/common/loadable';

const DynamicSecondPage = loadable(
  import('src/UI/pages/SecondPage/SecondPageConnector' /* webpackChunkName: "SecondPage" */),
)();

export default DynamicSecondPage;
