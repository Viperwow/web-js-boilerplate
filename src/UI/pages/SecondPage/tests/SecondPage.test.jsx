// Vendors
import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
// UI
import SecondPage from '..';

describe('SecondPage component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<SecondPage />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });
});
