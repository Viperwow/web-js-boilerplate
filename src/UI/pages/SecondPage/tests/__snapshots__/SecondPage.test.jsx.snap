// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondPage component should render correctly With default props 1`] = `
<div
  className="page"
>
  SECOND_PAGE_PLACEHOLDER
</div>
`;
